satpy/readers/agri_l1.py

```diff
@@ -143,11 +143,13 @@ def calibrate(self, data, ds_info, ds_name, file_key):
         elif calibration == 'radiance':
             raise NotImplementedError("Calibration to radiance is not supported.")
         # Apply range limits, but not for counts or we convert to float!
-        if calibration != 'counts':
+        if calibration != 'counts' and calibration is not None:
             data = data.where((data >= min(data.attrs['valid_range'])) &
                               (data <= max(data.attrs['valid_range'])))
         else:
             data.attrs['_FillValue'] = data.attrs['FillValue'].item()
+        if calibration is None:
+            data = data.where(data != data.attrs['FillValue'].item())
```

Comment:
Isn't calibration already None in the above `else` block so `FillValue` is handled twice...oh you're rewriting it to `_FillValue` but don't pop the original FillValue attribute? It feels like the `else` above and this `if calibration is None` block should be combined, right?

Reply:
Also done!